core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/samModels.scala

```diff
@@ -262,8 +263,68 @@ object PrivateAzureStorageAccountAction {
     sealerate.collect[PrivateAzureStorageAccountAction].map(a => (a.asString, a)).toMap
 }
 
-// TODO [IA-4608] merge with SamPolicyName
-/** Represents a role in Sam, permitting a set of actions on a resource of a certain type. */
+sealed trait RuntimeRole extends Product with Serializable {
+  def asString: String
+  override def toString = asString
+}
+object RuntimeRole {
+  final case object Creator extends RuntimeRole {
+    val asString = "creator"
+  }
+
+  final case object Manager extends RuntimeRole {
+    val asString = "manager"
+  }
+}
+
+sealed trait PersistentDiskRole extends Product with Serializable {
+  def asString: String
+  override def toString = asString
+}
+object PersistentDiskRole {
+  final case object Creator extends PersistentDiskRole {
+    val asString = "creator"
+  }
+
+  final case object Manager extends PersistentDiskRole {
+    val asString = "manager"
+  }
+}
+
+sealed trait AppRole extends Product with Serializable {
+  def asString: String
+  override def toString = asString
+}
+
+object AppRole {
+  final case object Creator extends AppRole {
+    val asString = "creator"
+  }
+
+  final case object Manager extends AppRole {
+    val asString = "manager"
+  }
+}
+
+sealed trait SharedAppRole extends Product with Serializable {
+  def asString: String
+  override def toString = asString
+}
+
+object SharedAppRole {
+  final case object Owner extends SharedAppRole {
+    val asString = "owner"
+  }
+
+  final case object User extends SharedAppRole {
+    val asString = "user"
+  }
+}
+
+/**
+ * Deprecated: use resource-type specific role enums (RuntimeRole, PersistentDiskRole, etc).
+ */
+@Deprecated
```

Comment:
We can remove `SamRole` and `SamPolicy` classes as of https://broadworkbench.atlassian.net/browse/IA-5063